packages/valory/skills/decision_maker_abci/policy.py

```diff
@@ -43,102 +43,138 @@ def argmax(li: List) -> int:
         """Deserialize a string to an `EGreedyPolicy` object."""
-        return EGreedyPolicy(**json.loads(policy))
+        return json.loads(policy, cls=EGreedyPolicyDecoder)
+
+    @property
+    def tools(self) -> List[str]:
+        """Get the number of the policy's tools."""
+        return list(self.accuracy_store.keys())
 
     @property
     def n_tools(self) -> int:
         """Get the number of the policy's tools."""
-        return len(self.counts)
+        return len(self.accuracy_store)
 
     @property
-    def random_tool(self) -> int:
-        """Get the index of a tool randomly."""
-        return random.randrange(self.n_tools)  # nosec
+    def n_requests(self) -> int:
+        """Get the total number of requests."""
+        return sum(acc_info.requests for acc_info in self.accuracy_store.values())
 
     @property
     def has_updated(self) -> bool:
         """Whether the policy has ever been updated since its genesis or not."""
-        return sum(self.counts) > 0
+        return self.n_requests > 0
 
     @property
-    def reward_rates(self) -> List[float]:
-        """Get the reward rates."""
-        return [
-            reward / count if count > 0 else 0
-            for reward, count in zip(self.rewards, self.counts)
-        ]
+    def random_tool(self) -> str:
+        """Get the name of a tool randomly."""
+        return random.choice(list(self.accuracy_store.keys()))  # nosec
 
     @property
-    def best_tool(self) -> int:
+    def best_tool(self) -> str:
         """Get the best tool."""
-        return argmax(self.reward_rates)
-
-    def add_new_tools(self, indexes: List[int], avoid_shift: bool = False) -> None:
-        """Add new tools to the current policy."""
-        if avoid_shift:
-            indexes = sorted(indexes, reverse=True)
-
-        for i in indexes:
-            self.counts.insert(i, self.initial_value)
-            self.rewards.insert(i, float(self.initial_value))
-
-    def remove_tools(self, indexes: List[int], avoid_shift: bool = False) -> None:
-        """Remove the knowledge for the tools corresponding to the given indexes."""
-        if avoid_shift:
-            indexes = sorted(indexes, reverse=True)
-
-        for i in indexes:
-            try:
-                del self.counts[i]
-                del self.rewards[i]
-            except IndexError as exc:
-                error = "Attempted to remove tools using incorrect indexes!"
-                raise ValueError(error) from exc
-
-    def select_tool(self, randomness: RandomnessType) -> Optional[int]:
+        weighted_accuracy = list(self.weighted_accuracy.values())
+        best = argmax(weighted_accuracy)
+        return self.tools[best]
+
+    def update_weighted_accuracy(self) -> None:
+        """Update the weighted accuracy for each tool."""
+        self.weighted_accuracy = {
+            tool: acc_info.accuracy
+            * (acc_info.requests - acc_info.pending)
```

Comment:
what is acc_info.pending?

Reply:
https://github.com/valory-xyz/trader/blob/b5562c28b0589e515c33efe4e00857e4ea828bc4/packages/valory/skills/decision_maker_abci/policy.py#L52